common/src/misc.rs

```diff
@@ -79,3 +85,308 @@ impl OutboundLocalTrust {
+/// This function calculates the start and end indices for a range of peers
+/// based on the provided `next_token` and `page_size`. The `next_token` is
+/// expected to be a base64 encoded string representing an 8-byte integer,
+/// which determines the starting peer index (`start_peer`). If `next_token`
+/// is `None`, the starting index defaults to 0.
+///
+/// The `end_peer` is calculated by adding the `page_size` (defaulting to 1000
+/// if not provided) to the `start_peer`. The `end_peer` is capped at
+/// `st_peers_cnt` to ensure it does not exceed the total number of peers.
+///
+/// # Arguments
+///
+/// * `st_peers_cnt` - The total number of seed trust peers.
+/// * `page_size` - Optional size of the page, determining the number of peers
+///   in the range.
+/// * `next_token` - Optional base64 encoded string used to determine the
+///   starting peer index.
+///
+/// # Returns
+///
+/// A `Result` containing a tuple with the start and end indices of the peer
+/// range, or a `BaseRunnerError` if decoding the `next_token` fails.
+pub fn compute_seedtrust_peer_range(
+    st_peers_cnt: u64, page_size: Option<usize>, next_token: Option<String>,
+) -> Result<(u64, u64), BaseRunnerError> {
+    let page_size = if let Some(page_size) = page_size { page_size as u64 } else { st_peers_cnt };
+
+    let start_peer = match next_token {
+        Some(token) => {
+            let decoded_bytes =
+                BASE64_STANDARD.decode(token).map_err(BaseRunnerError::Base64Decode)?;
+            let id_bytes = TryInto::<[u8; 8]>::try_into(decoded_bytes).map_err(|e| {
+                BaseRunnerError::Misc(format!("Failed to convert to 8 bytes: {:?}", e))
+            })?;
+            u64::from_be_bytes(id_bytes)
+        },
+        None => 0,
+    };
+    let start_peer = std::cmp::min(start_peer, st_peers_cnt);
+
+    let end_peer = start_peer + page_size;
+    let end_peer = std::cmp::min(end_peer, st_peers_cnt);
+
+    Ok((start_peer, end_peer))
+}
+
+/// Creates a next token for seed trust pagination.
+///
+/// This function generates a base64 encoded string that represents the `next_peer_id` as an 8-byte integer.
+/// The `next_peer_id` must be greater than 0 and less than `st_peers_cnt` to generate a non-empty token.
+///
+/// # Arguments
+///
+/// * `st_peers_cnt` - Total number of seed trust peers. It defines the upper limit for `next_peer_id`.
+/// * `next_peer_id` - The peer ID for which the next token is to be created.
+///
+/// # Returns
+///
+/// An `Option<String>` containing the base64 encoded next token if `next_peer_id` is in range; otherwise, `None`.
+pub fn create_seedtrust_next_token(st_peers_cnt: u64, next_peer_id: u64) -> Option<String> {
+    if next_peer_id == 0 || next_peer_id == st_peers_cnt {
+        None
+    } else {
+        let id_bytes = next_peer_id.to_be_bytes();
+        let next_token = BASE64_STANDARD.encode(id_bytes);
+        Some(next_token)
+    }
+}
+
+#[test]
+fn test_compute_localtrust_peer_range() {
```

Comment:
All the tests below this point should be under a `#[cfg(test)]` feature

Reply:
Done https://github.com/openrankprotocol/openrank/pull/220/commits/3d89f1434543170bb493cba729770fe5596e5681